### DISCUSSION TITLE
Watch on single input executing watch effect on a field array

### DISCUSSION THREAD

**Discussion by iamplo** [react-hook-form/react-hook-form#4329] [Q&A]
Hi, fantastic library thanks for all the effort put into it.

I'm trying to understand why an effect with a single input field `watch` dependency is also executing the effect for a `watch` dependency of an array field. See console log in:

https://codesandbox.io/s/watch-with-usefieldarray-forked-q436z?file=/src/App.tsx

1. Change the name field > observe both effects run in console log
2. Change an array input field > observe only the single effect run 

I also observed similar behavior with 2 fields arrays used in separate effects with their respective watch dependencies.

Thanks!

**Comment by bluebill1049** [comment]
The result of the `watch` is not memorized, it's meant to just use in the render.
